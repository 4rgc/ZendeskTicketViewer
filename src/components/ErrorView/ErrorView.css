.error-view-error,
.error-view-warning {
	border-radius: 10px;
	margin: 10px;
	display: flex;
	height: 30px;
	color: white;
	padding: 5px 10px;
	align-items: center;
}

.error-view-error {
	background-color: rgba(255, 0, 0, 0.3);
}

.error-view-warning {
	background-color: rgba(255, 165, 0, 0.3);
}

.error-view-message {
	margin-left: 10px;
}
