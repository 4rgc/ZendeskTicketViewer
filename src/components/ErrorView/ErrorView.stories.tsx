import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ErrorView } from './ErrorView';

export default {
	title: 'Components/ErrorView',
	component: ErrorView,
} as ComponentMeta<typeof ErrorView>;

const Template: ComponentStory<typeof ErrorView> = (args) => (
	<ErrorView {...args} />
);

export const ErrorType = Template.bind({});
ErrorType.args = {
	type: 'error',
	message: 'This is an error. Please contact support.',
};

export const WarningType = Template.bind({});
WarningType.args = {
	type: 'warning',
	message: 'This is a warning. Please try again later.',
};
