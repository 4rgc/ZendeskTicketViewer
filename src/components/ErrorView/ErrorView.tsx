import { ReactComponent as ErrorIcon } from '../../assets/error.svg';
import { ReactComponent as WarningIcon } from '../../assets/warning.svg';
import './ErrorView.css';

export type ErrorViewProps = {
	type: 'warning' | 'error';
	message: string;
};

export const ErrorView: React.FC<ErrorViewProps> = ({ type, message }) => {
	let errorViewClass = '';
	let Icon;
	switch (type) {
		case 'error':
			errorViewClass = 'error-view-error';
			Icon = <ErrorIcon fill="red" height="20px" width="20px" />;
			break;
		case 'warning':
			errorViewClass = 'error-view-warning';
			Icon = <WarningIcon fill="orange" height="20px" width="20px" />;
			break;
	}

	return (
		<div className={errorViewClass}>
			{Icon}
			<div className="error-view-message">{message}</div>
		</div>
	);
};
