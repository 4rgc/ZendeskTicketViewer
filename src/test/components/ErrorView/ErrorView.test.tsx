import { render } from 'enzyme';
import { ErrorView } from '../../../components/ErrorView/ErrorView';

describe('ErrorView tests', () => {
	it('should render the message as an error', () => {
		const error = render(
			<ErrorView
				type="error"
				message="This is an error. Please contact support"
			/>
		);
		expect(error).toMatchSnapshot();
	});

	it('should render the message as a warning', () => {
		const warning = render(
			<ErrorView
				type="warning"
				message="This is a warning. Please try again"
			/>
		);
		expect(warning).toMatchSnapshot();
	});
});
