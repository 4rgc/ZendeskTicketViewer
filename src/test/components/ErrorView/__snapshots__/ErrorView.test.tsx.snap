// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView tests should render the message as a warning 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "error-view-warning",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "fill": "orange",
          "height": "20px",
          "width": "20px",
        },
        "children": Array [
          Node {
            "data": "warning.svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Node {
          "attribs": Object {
            "class": "error-view-message",
          },
          "children": Array [
            Node {
              "data": "This is a warning. Please try again",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "error-view-message",
        },
        "children": Array [
          Node {
            "data": "This is a warning. Please try again",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "fill": "orange",
            "height": "20px",
            "width": "20px",
          },
          "children": Array [
            Node {
              "data": "warning.svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "fill": undefined,
            "height": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "fill": undefined,
            "height": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`ErrorView tests should render the message as an error 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "error-view-error",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "fill": "red",
          "height": "20px",
          "width": "20px",
        },
        "children": Array [
          Node {
            "data": "error.svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Node {
          "attribs": Object {
            "class": "error-view-message",
          },
          "children": Array [
            Node {
              "data": "This is an error. Please contact support",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "fill": undefined,
          "height": undefined,
          "width": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "error-view-message",
        },
        "children": Array [
          Node {
            "data": "This is an error. Please contact support",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "fill": "red",
            "height": "20px",
            "width": "20px",
          },
          "children": Array [
            Node {
              "data": "error.svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "fill": undefined,
            "height": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "fill": undefined,
            "height": undefined,
            "width": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
